Origin: commit 4b24b358cbc47e78e958e1789351a3db8be12eca
Author: Manuel A. Fernandez Montecelo <manuel.montezelo@gmail.com>
Date: Tue Aug 4 01:52:50 2015 +0100
Description: Fix boost::optional not converting to bool with gcc-5.2 and boost 1.58
 Fix problem with code which was working for many years (with boost 1.55 and many
 versions before), but now the straightforward option to convert boost::optional
 to bool does not work with gcc-5.2 and boost-1.58, so use .is_initialized()
 instead.

diff --git a/src/generic/problemresolver/search_graph.h b/src/generic/problemresolver/search_graph.h
index 0670f9b..50019fd 100644
--- a/src/generic/problemresolver/search_graph.h
+++ b/src/generic/problemresolver/search_graph.h
@@ -610,7 +610,15 @@ public:
   /** \brief Return \b true if this queue entry has a promotion and a
    *  successor.
    */
-  bool get_has_contents() const { return contents; }
+  bool get_has_contents() const {
+    // this code was working for many years (with boost 1.55 and many versions
+    // before), but now the straightforward option to convert boost::optional to
+    // bool does not work with gcc-5.2 and boost-1.58, so use .is_initialized()
+    // instead.
+
+    //return contents;
+    return contents.is_initialized();
+  }
 
   /** \brief Retrieve the promotion associated with this queue entry.
    */
